### PR TITLE
Openmp support

### PR DESCRIPTION
**Pull request by thilinarmtb**
Add OpenMP support so that we can use it for parallel algorithms.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,6 +87,20 @@ if (WITH_ARB)
     set(HAVE_SYMENGINE_ARB yes)
 endif()
 
+# OpenMp
+set(WITH_OPENMP no
+    CACHE BOOL "Build with OpenMP")
+
+if (WITH_OPENMP)
+    find_package(OpenMP REQUIRED)
```
Comment (certik) [inline CMakeLists.txt]: Is this `OpenMP` module distributed with cmake?

**Comment by certik** [comment]
This is good. But I think when openmp is enabled, we have to use the thread safe version of symengine.

File: CMakeLists.txt (hunk 1 above)
Comment (thilinarmtb) [inline CMakeLists.txt]: Yes.

**Comment by thilinarmtb** [comment]
I'll enable it.

**Comment by thilinarmtb** [comment]
Code block for finding OpenMP was placed after the line:

`set(WITH_SYMENGINE_THREAD_SAFE no CACHE BOOL "Enable SYMENGINE_THREAD_SAFE support")`

to avoid `SYMENGINE_THREAD_SAFE` being set to `no` for inconsistent user input
like the following:

`cmake -DWITH_OPENMP=yes WITH_SYMENGINE_THREAD_SAFE=no .`

**Comment by certik** [comment]
Very good, thanks! +1 to merge if tests pass.

**Comment by thilinarmtb** [comment]
Thanks for merging !!

**Pull request merged**